app/frontend/entrypoints/plates_from_tubes.js

```diff
@@ -98,4 +123,23 @@ $(() => {
       }
     }
   });
+
+  const form = $("#plates_from_tubes_form");
+  const hiddenMap = $("#plates_from_tubes_source_tubes_map")[0];
+  // We dynamically build barcode map on submit because we don't want to send the controller
+  // a messy string of barcodes and empty lines. We want a clean JSON map of well positions to barcodes.
+  form.on("submit", function () {
+    // Split by line break
+    const barcodes = editor.getValue().split(/\r?\n/);
```

Comment:
This may need to be tweaked depending on users input device. e.g. we need to confirm this still works with a scanner.